specification/network/resource-manager/Microsoft.Network/preview/2022-06-01-preview/networkManagerGroupMembership.json

```diff
@@ -0,0 +1,274 @@
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/providers/Microsoft.Network/networkGroupMemberships": {
+      "parameters": [
+        {
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "#/parameters/VirtualNetworkNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        },
+        {
+          "$ref": "#/parameters/FirstIndexParameter"
+        },
+        {
+          "$ref": "#/parameters/PageSizeParameter"
+        }
+      ],
+      "get": {
+        "tags": [
+          "NetworkGroupMemberships_List"
+        ],
+        "operationId": "ListNetworkGroupMemberships",
+        "description": "List networkGroupMemberships applied on a virtual network.",
+        "responses": {
+          "200": {
+            "description": "OK - Returns information about the configurations.",
+            "schema": {
+              "$ref": "#/definitions/NetworkGroupMembershipsListResult"
+            }
+          },
+          "default": {
+            "description": "Resource Provider error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List network group memberships": {
+            "$ref": "./examples/NetworkGroupMembershipsList.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "NetworkGroupMembershipsListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkGroupMembership"
+          },
+          "description": "Gets a page of NetworkGroupMembership"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets the URL to get the next page of results."
+        }
+      },
+      "description": "Result of the request to list NetworkGroupMembership."
+    },
+    "NetworkGroupMembership": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NetworkGroupMembershipProperties",
+          "description": "The network group membership properties"
+        },
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

Recommend replacing these top-level properties (id, location, name, type, systemData) with a $ref to TrackedResource definition in v4 common types.

Reply:
It is not trackedresource, and it doesn't include tags/etags. So we can't use reference